src/core/ddsi/src/q_entity.c

```diff
@@ -532,6 +582,61 @@ static void add_security_builtin_endpoints(struct participant *pp, ddsi_guid_t *
 }
 #endif
 
+
+#ifdef DDSI_INCLUDE_SECURITY
+
+static void connect_participant_secure(struct q_globals *gv, struct participant *pp)
+{
+  struct proxy_participant *proxypp;
+  struct ephash_enum_proxy_participant it;
+  struct ddsi_handshake *handshake;
+  nn_wctime_t timestamp;
+
+  if (q_omg_participant_is_secure(pp))
+  {
+    timestamp = now();
+    ephash_enum_proxy_participant_init (&it, gv->guid_hash);
+    while ((proxypp = ephash_enum_proxy_participant_next (&it)) != NULL)
+    {
+      /* Do not start handshaking when security info doesn't match. */
+      if ((proxypp->handshake_admin) && (q_omg_is_similar_participant_security_info(pp, proxypp)))
+      {
+        ddsi_hsadmin_lock(proxypp->handshake_admin);
+        handshake = ddsi_hsadmin_register_locked(proxypp->handshake_admin, pp, proxypp, timestamp, handshake_end_cb);
+        assert(handshake);
+        if (!handshake) {
```

Comment:
I'm pretty easy on the formatting, but to have different styles in a single function is a bit too much.

Secondly, if the ``assert(handshake)`` is correct, the test is superfluous; if the test is needed, then I'd be curious to know what will happen with this participant: surely without completing the handshake there will be no communication where some is expected/required and it gets to be really interesting if it succeeds for some but not all. Would it not make far more sense to refuse creation of the participant in such a case?

(And thirdly a small note: with a ``struct q_globals *gv`` in scope, you can use ``GVWARNING`` which is a bit easier.)